nginxcfg/nginxcfg.py

```diff
@@ -0,0 +1,255 @@
+            server_dict = {}
+            stored = ''
+            for line_num, _ in enumerate(vhost_data):
+                if line_num >= server_block[0]:
+                    l = vhost_data[line_num]
+                    if line_num >= server_block[1]:
+                        server_dict['alias'] = alias
+                        server_dict['l_num'] = server_block[0]
+                        server_dict['config_file'] = config_file
+                        server_dict['ip_port'] = ip_port
+                        server_dict_ret.append(server_dict)
+                        server_name_found = False
+                        break
+
+                    if l.startswith('#'):
+                        continue
+                    l = l.split('#')[0]
+
+                    if not l.strip().endswith(';'):
+                        if line_num != server_block[0]:
+                            stored += l.strip() + ' '
+                    else:
+                        l = stored + l
+                        l = l.strip().strip(';')
+                        stored = ''
+
+                    if l.startswith('server_name') and server_name_found:
+                        alias += l.split()[1:]
+
+                    if l.startswith('server_name'):
+                        if l.split()[1] == "_":
+                            server_dict['servername'] = "default_server_name"
+                        else:
+                            server_dict['servername'] = l.split()[1]
+                        server_name_found = True
+                        if len(l.split()) >= 2:
+                            alias += l.split()[2:]
+                    if l.startswith('listen'):
+                        ip_port.append(l.split()[1])
+        return server_dict_ret
+
+    def get_vhosts(self, exclude_string):
+        vhosts_list = self._get_vhosts()
+        print("{0}nginx vhost configuration:{1}".format(
+            COLOURS["WHITE"], COLOURS["RESET"]))
+        for vhost in vhosts_list:
+            ip_ports = vhost['ip_port']
+            for ip_port_x in ip_ports:
+                if '[::]' in ip_port_x:
+                    pattern = re.compile(r'(\[::\]):(\d{2,5})')
+                    pattern_res = re.match(pattern, ip_port_x)
+                    ip = pattern_res.groups()[0]
+                    port = pattern_res.groups()[1]
+                else:
+                    ip_port = ip_port_x.split(':')
+                    try:
+                        ip = ip_port[0]
+                        port = ip_port[1]
+                    except:  # pylint: disable=bare-except
+                        ip = '*'
+                        port = ip_port[0]
+                servername = vhost.get('servername', None)
+                serveralias = vhost.get('alias', None)
+                line_number = vhost.get('l_num', None)
+                config_file = vhost.get('config_file', None)
+                try:
+                    if exclude_string in servername:
```

Comment:
The `exclude_string` is only active on the  `servername` ("1 per Server block") and not for the other `alias` names on a Server block.

Is this intentional - does it need better documenting?